src/Microsoft.ML.Data/Evaluators/Metrics/AnomalyDetectionMetrics.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Data.DataView;
+
+namespace Microsoft.ML.Data.Evaluators.Metrics
+{
+    /// <summary>
+    /// Evaluation results for anomaly detection.
+    /// </summary>
+    public sealed class AnomalyDetectionMetrics
+    {
+        /// <summary>
+        /// Gets the area under the ROC curve.
+        /// </summary>
+        /// <remarks>
+        /// The area under the ROC curve is equal to the probability that the algorithm ranks
+        /// a randomly chosen positive instance higher than a randomly chosen negative one
+        /// (assuming 'positive' ranks higher than 'negative').
+        /// </remarks>
+        public double Auc { get; }
+
+        /// <summary>
+        /// Detection rate at k false positives.
+        /// </summary>
+        public double DrAtK { get; }
+        /// <summary>
+        /// Detection rate at fraction p false positives.
```

Comment:
>p [](start = 39, length = 1)

What is `p`?

Reply:
to me the Detection rate metrics look suspicious.  perhaps we should not expose these for V1.0 .  only expose AUC

---
In reply to: [255719011](https://github.com/dotnet/machinelearning/pull/2390#discussion_r255719011) [](ancestors = 255719011)

Reply:
removed that detectionrate@p metric.

---
In reply to: [255725823](https://github.com/dotnet/machinelearning/pull/2390#discussion_r255725823) [](ancestors = 255725823,255719011)